### PR TITLE
Convert BaseStorageIntegrationTest to a JUnit 5 extension

### PR DESCRIPTION
**Pull request by dimas-b**
* Better control over test lifecycle.

* Does not require extending any particular base test class.

### CODE REVIEW

**Comment by dimas-b** [comment]
Let's merge this part now to enable repo reorg. I will do another PR for follow-up test improvements

File: testing/src/test/java/io/stargate/it/storage/ExternalStorage.java [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+  private static final Logger LOG = LoggerFactory.getLogger(ExternalStorage.class);
+
+  private static final String CCM_VERSION = "ccm.version";
+  private static final boolean EXTERNAL_BACKEND =
+      Boolean.getBoolean("stargate.test.backend.use.external");
+
+  private static final AtomicBoolean executing = new AtomicBoolean();
+
+  private static final int clusterNodes = Integer.getInteger("stargate.test.backend.nodes", 1);
```
Comment (dougwettlaufer) [inline testing/src/test/java/io/stargate/it/storage/ExternalStorage.java:45]: How flexible will this be for the tests @tomekl007 is working on where he needs multiple nodes?

File: testing/src/test/java/io/stargate/it/storage/ExternalStorage.java [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** JUnit 5 extension for tests that need a backend database cluster managed by {@code ccm}. */
+public class ExternalStorage implements BeforeAllCallback, AfterAllCallback, ParameterResolver {
```
Comment (dougwettlaufer) [inline testing/src/test/java/io/stargate/it/storage/ExternalStorage.java:35]: In the future how hard will it be to include non-ccm managed storage?

**Review by dougwettlaufer — APPROVED** [review]
Looks good just a few minor questions.

File: testing/src/test/java/io/stargate/it/storage/ExternalStorage.java (hunk 1 above)
Comment (dimas-b) [reply testing/src/test/java/io/stargate/it/storage/ExternalStorage.java:45]: Ah, I actually missed this one - it's supposed to come from `ClusterSpec`.

My thinking is that each class will define how many nodes it needs. We could support per-method specs too.

Do we foresee setting cluster size from outside of the test code scope (e.g. in maven properties / run config)?

File: testing/src/test/java/io/stargate/it/storage/ExternalStorage.java (hunk 2 above)
Comment (dimas-b) [reply testing/src/test/java/io/stargate/it/storage/ExternalStorage.java:35]: Just update this class to support it.

Comment (dimas-b) [reply testing/src/test/java/io/stargate/it/storage/ExternalStorage.java:35]: We could use different `ClusterSpec` -like annotations too.

**Comment by dimas-b** [comment]
Merging now to unblock repo reorg. I'll fix the cluster size handling in a follow-up RP.

File: testing/src/test/java/io/stargate/it/storage/ExternalStorage.java (hunk 1 above)
Comment (dougwettlaufer) [reply testing/src/test/java/io/stargate/it/storage/ExternalStorage.java:45]: No, I'd imagine cluster sizes will be determined internally by the tests themselves.

File: testing/src/test/java/io/stargate/it/storage/ExternalStorage.java (hunk 2 above)
Comment (dougwettlaufer) [reply testing/src/test/java/io/stargate/it/storage/ExternalStorage.java:35]: Got it, thanks.

**Pull request merged**